Builds fail using Dockerfile

Reported by Chris in chat:

```
Log Tail:       [1/2] STEP 7/7: RUN make &&         mv bridge /tmp/                                                                                                                      
                gcc -MT obj/amqp_rcv_th.o -MD -MP -MF obj/.deps/amqp_rcv_t...tion -fcf-protection  -c -o obj/amqp_rcv_th.o amqp_rcv_th.c                                                 
                gcc: error: /usr/lib/rpm/redhat/redhat-hardened-cc1: No such file or directory                                                                                           
                make: *** [Makefile:70: obj/amqp_rcv_th.o] Error 1                                                                                                                       
                error: build error: building at STEP "RUN make &&         mv bridge /tmp/": while running runtime: exit status 2                                                         
```

Issue due to changes in #29

--- a/.github/workflows/linter.yml
+++ b/.github/workflows/linter.yml
@@ -28,3 +28,12 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  docker-build-check:
+    name: Check Dockerfile results in successful build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Check build of Dockerfile is successful
+        run: docker build -t sg-bridge:check-build -f build/Dockerfile .

--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 
 MAJOR?=0
 MINOR?=1
- 
+
 VERSION=$(MAJOR).$(MINOR)
 
 BIN := bridge
@@ -22,7 +22,7 @@ $(shell mkdir -p $(dir $(OBJS)) >/dev/null)
 $(shell mkdir -p $(dir $(DEPS)) >/dev/null)
 
 CC=gcc
-CFLAGS+=-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
+CFLAGS+=-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
 LDLIBS=-lqpid-proton -lpthread
 LDFLAGS+=-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1
 

--- a/build/Dockerfile
+++ b/build/Dockerfile
@@ -4,8 +4,11 @@ FROM registry.access.redhat.com/ubi8 AS builder
 # dependencies for qpid-proton-c
 COPY build/repos/opstools.repo /etc/yum.repos.d/opstools.repo
 
+# redhat-rpm-config is required to provide hardening compiling instructions
+# (such as /usr/lib/rpm/redhat/redhat-hardened-cc1) even though we're not
+# building RPMs here
 RUN dnf install qpid-proton-c-devel --setopt=tsflags=nodocs -y && \
-        dnf install gcc make -y && \
+        dnf install gcc make redhat-rpm-config -y && \
         dnf clean all
 
 ENV D=/home/bridge

